src/tt.cpp

```diff
@@ -61,12 +65,17 @@ void TTEntry::save(Key k, Value v, bool pv, Bound b, Depth d, Move m, Value ev)
 
 void TranspositionTable::resize(size_t mbSize) {
 
+  size_t allocSize;
+
   Threads.main()->wait_for_search_finished();
 
   clusterCount = mbSize * 1024 * 1024 / sizeof(Cluster);
 
   free(mem);
-  mem = malloc(clusterCount * sizeof(Cluster) + CacheLineSize - 1);
+  allocSize = clusterCount * sizeof(Cluster) + CacheLineSize - 1;
+
+  // align by 2 MB to make the allocation huge page friendly
+  mem = aligned_alloc(2 * 1024 * 1024, allocSize);
```

Comment:
aligned_alloc() is not supported on Windows, it seems. Funny, since that's C11 standard function... I'll add a workaround.

Reply:
did you include the needed header files? Better include explicitly anyway.

Reply:
Both aligned_alloc() and malloc() come from <cstdlib>, which is included by "types.h". I can revise this patch and add the include explicitly.

Google tells me that aligned_alloc() is apparently missing from VS. There are some alternatives, but those are Windows-specific: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/aligned-malloc?view=vs-2019 

Anyways, I don't think 2MB page alloc is really needed unless we have huge pages, so malloc() should be good enough. If we want huge pages on Windows, the instructions are here: https://docs.microsoft.com/en-us/windows/win32/memory/large-page-support . SeLockMemoryPrivilege sounds like regular users don't have this privilege, and I don't have a Windows development environment handy for experimenting.